### PR TITLE
Disable update button when the selected value is null

### LINKED ISSUE #11096: [Bug]: Project PM UI Install button should disable when `null` version selected
### NuGet Product Used

Visual Studio Package Management UI

### Product Version

VS any

### Worked before?

_No response_

### Impact

It bothers me. A fix would be nice

### Repro Steps & Context

Due to a bug in WPF https://github.com/dotnet/wpf/issues/4929,
..NuGet sometimes allows selection of a null separator https://github.com/nuget/home/issues/11007 (see for repro steps).

In Solution PMUI, the Install button is disabled as expected.
Project PM UI is not disabled, but should be. Note that the action does not occur, but no info is given to the user.

image

### Verbose Logs

_No response_

### PR DESCRIPTION
**Pull request by martinrrm**
## Bug 

Fixes: https://github.com/NuGet/Home/issues/11096

Regression? Last working version:

## Description

Due to a bug in WPF dotnet/wpf#4929, NuGet sometimes allows selection of a null separator https://github.com/nuget/home/issues/11007. This pr disables the button when the SelectedVersion is null

image

## PR Checklist

- [ ] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [ ] N/A

### CODE REVIEW

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/ProjectView.xaml
Comment (erdembayar) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/ProjectView.xaml]: Do we need `Mode=OneWay` here as well?
Did you test if keep switch back and forth by selecting and unselecting value also reflected on `Enabld/Disabled` mode of button?

Comment (martinrrm) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/ProjectView.xaml]: I added the `Mode=OneWay` and yes I tested that. Also found that this error happened in the Install button so I added the same conditional.

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Models/PackageDetailControlModel.cs
Comment (zivkan) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Models/PackageDetailControlModel.cs]: Generally, properties and methods should have positive names, to avoid confusion over double negatives. I realize this is to use the `NotNullOrTrueToBooleanConverter`.  Maybe it's not worthwhile, but would a new converter be useful?

Honestly, by the name alone, I really struggle to guess what `NotNullOrTrueToBooleanConverter` is expected to do. It sounds like it returns `value == null || value == true`. Therefore, if your property was `IsInstalledEnabled`, or `IsInstalledButtonEnabled`, then you don't need a converter?

Comment (zivkan) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/Models/PackageDetailControlModel.cs]: Thank you for the change. It's actually easy to understand now.

**Review by zivkan — APPROVED** [review]
I'd feel more comfortable if you get approval from someone who has more XAML experience than me before merging.

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Models/PackageDetailControlModel.cs
Comment (donnie-msft) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Models/PackageDetailControlModel.cs:222]:
```suggestion
        public bool IsInstallOrUpdateButtonEnabled
```

..or maybe..

```suggestion
        public bool CanSelectedVersionBeInstalled
```

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/ProjectView.xaml
Comment (donnie-msft) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/ProjectView.xaml:152]: ✅ Binding makes sense as OneWay to me.

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Models/PackageDetailControlModel.cs
Comment (donnie-msft) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Models/PackageDetailControlModel.cs:208]: You can remove all property changed for `IsSelectedVersionInstalled`, because as I can tell, it's no longer bound to anything.

**Review by donnie-msft — CHANGES_REQUESTED** [review]
A simple request and I think it's good to go.

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/ProjectView.xaml
Comment (kartheekp-ms) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/ProjectView.xaml:140]:
``` suggestion
      IsEnabled="{Binding IsSelectedVersionInstalled, Converter={StaticResource InverseBooleanConverter}, Mode=OneWay}"
```
I am a beginner in WPF and have limited knowledge on how bindings work. If I understand the logic correctly, the button is enabled when `IsSelectedVersionInstalled = false`. How about using the existing `InverseBooleanConverter` which returns `true` when `IsSelectedVersionInstalled` returns `false`? If this suggestion is acceptable then we can avoid creating new property. This comment applies to line no. `152`.

Comment (jeffkl) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/ProjectView.xaml:140]: This looks cleaner to me.  If it works, I'd say use this instead of adding a new property.

**Review by jeffkl — APPROVED** [review]

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/ProjectView.xaml
Comment (martinrrm) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/ProjectView.xaml:140]: What I'm trying to fix here is also show the button disabled when the selected version is null, that's why I'm creating a new variable to avoid changing/removing `IsSelectedVersionInstalled`, I'm going to move this code to this [pr](https://github.com/NuGet/NuGet.Client/pull/4355) because with my new feature ([Custom Versions](https://github.com/NuGet/NuGet.Client/pull/4339)) we can have nulls in `SelectedVersion` more often

**Comment by ghost** [comment]
This PR has been automatically marked as stale because it has no activity for **7 days**. It will be closed if no further activity occurs **within another 7 days of this comment**. If it is closed, you may reopen it anytime when you're ready again, as long as you don't delete the branch.

**Comment by ghost** [comment]
This PR has been automatically marked as stale because it has no activity for **7 days**. It will be closed if no further activity occurs **within another 7 days of this comment**. If it is closed, you may reopen it anytime when you're ready again, as long as you don't delete the branch.

**Pull request closed without merging**
